service/internal/logger/audit/serviceInterceptor.go

```diff
@@ -0,0 +1,33 @@
+package audit
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// The audit unary server interceptor is a gRPC interceptor that adds metadata
+// to the context of incoming requests. This metadata is used to log audit events
+func UnaryServerInterceptor(ctx context.Context, req any, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
+	// Generate and set a request ID on incoming requests
+	requestUUID := uuid.New()
+	ctx = context.WithValue(ctx, RequestIDContextKey, requestUUID)
+
+	// Get metadata from the context
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Error(codes.Unauthenticated, "missing metadata")
```

Comment:
Are there any public unauthenticated routes this interceptor expects to hit? If so, is there a more accurate code than unauthenticated? https://grpc.io/docs/guides/status-codes/#the-full-list-of-status-codes

Reply:
Yeah I guess I just defaulted to this because I originally had this in the authn.go file before I decided it didn't fit there that well. I would imagine the best code from that list might be `INVALID_ARGUMENT`?

Reply:
I think that makes sense. It's the equivalent of a 400 in gRPC.

Reply:
Updated to INVLAID_ARGUMENT